File: LC/Python/Medium/424.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        seen = defaultdict(int)
        ans = 0
        window_start = 0
        
        for window_end in range(len(s)):
            seen[s[window_end]] += 1
            
            if window_end - window_start + 1 - max(seen.values()) > k:
                seen[s[window_start]] -= 1
                window_start += 1
            
            ans = max(ans, window_end - window_start + 1)
        
        return ans

File: LC/C++/Medium/424.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int ans = 0;
        std::unordered_map<char, int> seen{};
        int l = 0;
        int maxSeen = 0;

        for (int r{}; r < s.size(); ++r) {
            maxSeen = std::max(maxSeen, ++seen[s[r]]);
            while (r - l + 1 - maxSeen > k) {
                seen[s[l]]--;
                l++;
            }
            ans = std::max(ans, r - l + 1);
        }

        return ans;
    }
};